sonaranalyzer-dotnet/src/SonarAnalyzer.CFG/ControlFlowGraph/CSharpControlFlowGraphBuilder.cs

```diff
@@ -471,12 +471,7 @@ private Block BuildExpression(ExpressionSyntax expression, Block currentBlock)
                     return BuildJumpToExitStatement(throwExpression, currentBlock, throwExpression.Expression);
 
                 case SyntaxKindEx.DeclarationExpression:
-                    // we ignore 'discard' parameters
-                    var declaration = (DeclarationExpressionSyntaxWrapper)expression;
-                    if (!declaration.Designation.SyntaxNode.IsKind(SyntaxKindEx.DiscardDesignation))
-                    {
-                        currentBlock.ReversedInstructions.Add(expression);
-                    }
```

Comment:
This will affect also **sonar-security**. I will do a separate commit for handling the update.